yt_dlp/extractor/jupiter.py

```diff
@@ -0,0 +1,59 @@
+from .common import InfoExtractor
+from .. import int_or_none
+from ..utils import (
+    traverse_obj,
+)
+
+
+class JupiterIE(InfoExtractor):
+    _VALID_URL = r'https://jupiter\.err\.ee/(?P<id>\d+)/'
+    _TESTS = [{
+        'note': 'S01E06: Impulss',
+        'url': 'https://jupiter.err.ee/1609145945/impulss',
+        'md5': '1ff59d535310ac9c5cf5f287d8f91b2d',
+        'info_dict': {
+            'id': '1609145945',
+            'ext': 'mp4',
+            'title': 'Loteriipilet hooldekodusse',
+            'description': 'md5:fa8a2ed0cdccb130211513443ee4d571',
+            'upload_date': '20231107',
+            'timestamp': 1699380000,
+            'series': 'Impulss',
+            'season': 'Season 1',
+            'season_number': 1,
+            'episode': 'Loteriipilet hooldekodusse',
+            'episode_number': 6,
+            'series_id': 1609108187,
+            'release_year': 2023,
+            'episode_id': 1609145945,
+        },
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        content_url = f"https://services.err.ee/api/v2/vodContent/getContentPageData?contentId={video_id}"
+        data = traverse_obj(self._download_json(content_url, video_id), ('data', 'mainContent'))
```

Comment:
> data extraction should also be fatal.

what do you mean? do you suggest a code for it? wrap with `try_call`?

currently it can fail with:
1. 404 response (I modified URL a bit)
2. array dereferencing (I modified key "data" to "blah"

So submitted something: de4fdeea6, not sure about error text.

Reply:
I just meant to access the dict directly through `['data']['mainContent']` instead of using `traverse_obj`.
The data extraction should be fatal since we need `data` to proceed anyway. So there is no need for `try_call`.

Reply:
followup is here:
- https://github.com/yt-dlp/yt-dlp/pull/8549#discussion_r1466201121